frontend/src/store/course/actions.ts

```diff
@@ -169,21 +169,22 @@ const chooseMentor = createAsyncThunk<
   return;
 });
 
-const updateisMentorChoosingEnabled = createAsyncThunk<
+const updateIsMentorChoosingEnabled = createAsyncThunk<
   boolean,
-  void,
+  CourseGetRequestParamsDto,
   AsyncThunkConfig
->(ActionType.SET_IS_MENTOR_CHOOSING_ENABLED, (_, { getState }) => {
-  const {
-    auth: { user },
-    course: { mentors },
-  } = getState();
+>(ActionType.SET_IS_MENTOR_CHOOSING_ENABLED, async ({ id }, { extra }) => {
+  const { coursesApi } = extra;
 
-  const isMentorCheck = mentors.some(
-    (mentor) => mentor.id === (user as UserWithPermissions).id,
-  );
+  const isMentorCheck = await coursesApi.checkIsMentor({
+    courseId: id,
+  });
+  const hasMentorCheck = await coursesApi.checkHasMentor({
+    courseId: id,
+  });
+  const check = isMentorCheck || hasMentorCheck;
```

Comment:
Don't know what `check` means
```suggestion
  const isMentorChoosingEnabled = isMentorCheck || hasMentorCheck;
```